### ISSUE TITLE
Fetch feed once for multiple install targets

### ISSUE THREAD

**Issue by jayvdb** [chocolatey/choco#1840]
When installing multiple packages, the NuGet feed at https://chocolatey.org/api/v2/ is loaded for each package.  

I recently had one package install, and then the others each failed noisily while loading the feed multiple times.

It would be nice if the feed is loaded once, the list of package name requested checked once, and metadata saved.

It would also be great if there was a 'download-first' option, to do all the fetching first (and parallelise it! ;-)) before starting any installs.

```console
cinst -y rust R.project ghc haskell-stack julia luarocks
Chocolatey v0.10.11
Installing the following packages:
rust;R.project;ghc;haskell-stack;julia;luarocks
By installing you accept licenses for the packages.
Progress: Downloading rust 1.35.0.20190524... 100%
....
The install of rust was successful.
  Software installed to 'C:\ProgramData\chocolatey\lib\rust\tools'

[NuGet] Could not connect to the feed specified at 'https://chocolatey.org/api/v2/'. Please verify that the package source (located in the Package Manager Settings) is valid and ensure your network connectivity.
R.project not installed. The package was not found with the source(s) listed.
 Source(s): 'https://chocolatey.org/api/v2/'
 NOTE: When you specify explicit sources, it overrides default sources.
If the package version is a prerelease and you didn't specify `--pre`,
 the package may not be found.
Please see https://chocolatey.org/docs/troubleshooting for more 
 assistance.
(repeat)
(repeat)
(repeat)
(repeat)
(repeat)
```

https://ci.appveyor.com/project/jayvdb/package-manager/builds/24997168/job/hre68y5hjwv9qmfg

**Comment by ferventcoder** [comment]
Duplicate of #820

**Comment by ferventcoder** [comment]
Howdy @jayvdb - We agree, and to do so we'll need to own more of the nuget architecture that handles this.

**Comment by ferventcoder** [comment]
As far as the issue you are seeing, any idea why it is failing? It appears that AppVeyor or the community repository may have ran into some issues somewhere causing the rest of the calls to fail. This may also have an issue logged for it, but if you want to explore the failures, we can either rewrite and reopen this issue or start a new one.

**Comment by jayvdb** [comment]
I dont know the cause, but I'm similar problems annoyingly frequently while working on converting to a choco based CI - https://gitlab.com/coala/package_manager/merge_requests/168 is the MR.  These are quite heavy packages to install, so that might be related.

Since it doesnt look like there is real interest in implement any of what I requested in this issue soon, I'll add a --whatif invocation first.  Maybe that will help.

**Comment by jayvdb** [comment]
> we'll need to own more of the nuget architecture that handles this.

Why cant you download the feed once at the beginning of invocation, and then give it to NuGet each time?

Are their NuGet features missing that you need.

#820 is a literally empty issue with a vague title, so it is hard to see how my issue with concrete feature requests is a duplicate of it.
